### PR TITLE
Add a documentation style guide

### PR DESCRIPTION
**Pull request by pavithraes**
## Reference Issues or PRs

Fixes: #153

## What does this implement/fix?

_Put an `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features to not work as expected)
- [x] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [ ] Did you test the pull request locally?
- [ ] Did you add new tests?

#### Text styling

- [x] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [x] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [x] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [x] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If the content were to be read as plain text, it still makes sense, and no information is missing.

### CODE REVIEW

File: docs/docs/community/style-guide.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+id: style-guide
+title: Documentation style guide
+description: Guidelines for writing consistent technical documentation.
+---
+
+As an open source project, we understand the many people with a variety of different backgrounds contribute to Nebari's documentation. A style guide helps us maintain consistency in the language used and allows us to clearly communicate our accessibility standards. We want all our users to find the documentation helpful and get the information they need without hurdles.
+
+Nebari's documentation follows the [Google developer documentation style guide](https://developers.google.cn/style). This style guide is very thorough, and therefore very large. We have a shorter list of guidelines in the following sections for a quick reference.
+
```
Comment (pavithraes) [inline docs/docs/community/style-guide.md:10]:
```suggestion

These are not rules, they are suggestions. Follow your best judgment and feel free to add to this document!

```

Comment (pavithraes) [inline docs/docs/community/style-guide.md:10]:
```suggestion

These are not rules, they are suggestions. Follow your best judgment and feel free to add to this document!

```

**Review by trallard — CHANGES_REQUESTED** [review]
This looks excellent @pavithraes - left some suggestions and notes, but otherwise this seems so close to merge 🚀 🌻

File: docs/docs/community/style-guide.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+- Always use **text to describe where links go**. For example, instead of "read more", write "read more about accessible writing in the style guide".
+- Take the time to **write good alt-text** for images, diagrams, and graphics. The alt text should describe the setting (for example, "A JupyterLab window showing ...") as well as the overarching message that the graphic conveys (for example, "... denotes consistency across the Dask dashboard plots"). Read the [Text associated with images section of the Google Developer Style Guide](https://developers.google.com/style/images#text-associated-with-images) for more details.
+- Practice **\*inclusive and bias-free communication**. Learn more in:
+  - [Write inclusive documentation, Google developer documentation style guide](https://developers.google.com/style/inclusive-documentation)
+  - [Bias-free communication, Microsoft writing style guide](https://learn.microsoft.com/en-gb/style-guide/bias-free-communication)
+  - [Gender identity, Apple style guide](https://support.apple.com/en-gb/guide/applestyleguide/apd2a7af8d36/web)
+
```
[the remark this answers is not in the document]
Comment (pavithraes) [reply docs/docs/community/style-guide.md:47]: Thank you!

File: docs/docs/community/style-guide.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+| Panel            | First letter always capitalized                                                          |
+| VS Code          | 'V', 'S', and 'C' always capitalized; always use one blank space between 'VS' and 'Code' |
+| Jupyter          | First letter always capitalized                                                          |
+| Jupyter Notebook | 'J' and 'N' always capitalized                                                           |
+| JupyterLab       | 'J' and 'L' always capitalized                                                           |
+| Kubernetes       | First letter always capitalized                                                          |
+| Docker           | First letter always capitalized                                                          |
+| Terraform        | First letter always capitalized                                                          |
+| YAML | Always capitalized |
+
```
[the remark this answers is not in the document]
Comment (pavithraes) [reply docs/docs/community/style-guide.md:77]: I see it's always lowercase in our docs + branding material, so I'll go with that. I'll also make sure to confirm with Dharhas and Pierre. :)

**Review by trallard — APPROVED** [review]
Perfect! Merging

**Pull request merged**